query/src/main/java/org/infinispan/query/impl/SearchManagerImpl.java

```diff
@@ -148,8 +146,7 @@ public void purge(Class<?> entityType) {
 
    @Override
    public <T> T unwrap(Class<T> cls) {
-      if (SearchIntegrator.class.isAssignableFrom(cls) || ExtendedSearchIntegrator.class.isAssignableFrom(cls)
-            || SearchFactoryState.class.isAssignableFrom(cls)) {
+      if (SearchIntegrator.class.isAssignableFrom(cls)) {
```

Comment:
By snapshot I mean adding required stuff to SearchIntegrator, build HSEARCH
x.y.z-SNAPSHOT, use that version in ISPN and test. If it's good, a release
can be done with the new functionality, otherwise it can be incrementally
improved until it's fine.